spec/controllers/forgot_passwords_controller_spec.rb

```diff
@@ -0,0 +1,47 @@
+require 'spec_helper'
+
+describe ForgotPasswordsController do
+  describe "POST create" do
+    context "with valid email address" do
+      it "generates a token for the user" do
+        bob = Fabricate :user, email: "bob@bob.com"
+        expect(bob.token).to be_nil
+        post :create, email: "bob@bob.com"
+        expect(bob.reload.token).to be_present
+      end
+
+      it "sends an email with a link to reset that user's password" do
+        bob = Fabricate :user, email: "bob@bob.com"
+        post :create, email: "bob@bob.com"
+        email_message = ActionMailer::Base.deliveries.last
+        expect(email_message).to be_present
+        expect(email_message.to).to eq(["bob@bob.com"])
+        ActionMailer::Base.deliveries.clear
+      end
+
+      it "renders password-reset confirmation page" do
```

Comment:
You should clear email deliveries here as well. They are there even tho you are rendering the template only. This _can_ and _will_ cause issues with other specs working with emails.